doc/dev/design/FABRID.rst

```diff
@@ -0,0 +1,471 @@
+features source authentication for routers and the destination host, and path validation for source and destination hosts.
+
+Proposal
+========
+
+FABRID indroduces policies, which can be thought of as additional path constraints such that the ASes only use intra-AS paths that fulfill that policy.
+The ASes announce those policies to the end hosts, who then can use those additional path constraint during path selection.
+The border routers use those policies to decide on the intra-AS path to forward, e.g. by using MPLS labels.
+Some FABRID policies are globally defined and others locally per AS.
+Global policies make sense in order to have known policies for the common use cases where each end host knows that if an AS supports that policy,
+then it knows the constraints of that policy without the need of fetching the full policy description.
+The AS network operator configures which global FABRID policies are supported for the local AS and can add additional local FABRID
+policies that are valid for this AS.
+The FABRID policy identifiers are provided to the endhosts inside of the PCB or as an detachable PCB extension.
+The FABRID policy descriptions on the other hand have to be requested from the local control service which will then either return
+the cached FABRID policy description or query the policy description from the remote AS.
+That way we can minimize the overhead that the FABRID polices cause to the PCB size.
+A source endhost can then select a path together with FABRID policies and forward the FABRID packet over this path to a destination endhost.
+The on-path border routers will then add some proof of transit such that the destination endhost can later verify that the packet indeed followed the intended path.
+
+Our proposed design and implementation of FABRID allows for incremental deployment at router- and AS-level, i.e., some AS operators may want to
+deploy FABRID while others do not, and those who do may only want to deploy it on a subset of border routers.
+This allows for a smooth migration where an AS can test-wise update some border routers and test that nothing breaks.
+However, this could lead to the situation where an end host does not have paths available with all on-path ASes supporting FABRID.
+In such a situation the end host can still send its traffic along that path, but the guarantees provided by FABRID only hold for the FABRID-enabled ASes.
+
+Since each AS can create their own local FABRID policies, end hosts have to retrieve them.
+In our design, end hosts have to retrieve them from their local AS, and the local AS has to retrieve them from the remote AS, similar to how SCION path retrieval is implemented.
+Those policies are only fetched on demand by the local control service and will be cached till end of their validity.
+This allows for better scalability for the FABRID policies because an AS does not have to retrieve all FABRID policies from all other ASes.
```

Comment:
This on-demand policy retrieval improves the scalability of the control service compared to simply fetching all the policies from all ASes in the Internet.